apps/internal/oauth/ops/authority/authority.go

```diff
@@ -351,29 +352,31 @@ type Info struct {
 	InstanceDiscoveryDisabled bool
 }
 
-func firstPathSegment(u *url.URL) (string, error) {
-	pathParts := strings.Split(u.EscapedPath(), "/")
-	if len(pathParts) >= 2 {
-		return pathParts[1], nil
-	}
-
-	return "", errors.New(`authority must be an https URL such as "https://login.microsoftonline.com/<your tenant>"`)
-}
-
 // NewInfoFromAuthorityURI creates an AuthorityInfo instance from the authority URL provided.
 func NewInfoFromAuthorityURI(authority string, validateAuthority bool, instanceDiscoveryDisabled bool) (Info, error) {
 	u, err := url.Parse(strings.ToLower(authority))
 	if err != nil || u.Scheme != "https" {
 		return Info{}, errors.New(`authority must be an https URL such as "https://login.microsoftonline.com/<your tenant>"`)
 	}
 
-	tenant, err := firstPathSegment(u)
```

Comment:
The authority should always be a valid URL. I would parse it into standard http object and then did the test for certain authorities in that domain (on http level). I think it will make it clear what properties the tenant URL has to have to be considered a certain type (ADFS/DSTS).